### PR TITLE
fix: guard entity identity corruption and reject abstract items (#339)

### LINKED ISSUE #339: bug: entity identity corruption and abstract concepts classified as physical items
## Summary

Entity detail updates from the LLM sometimes overwrite an entity's core identity with unrelated content, or classify abstract concepts/methods as physical items. This produces entities with internally contradictory data.

### Identity Corruption (wrong entity overwritten)

**`item-spear`** — A spear from turn-011 had its name and identity overwritten with "sturdy bowl" data:

```json
{
  "id": "item-spear",
  "name": "sturdy bowl",
  "identity": "A practical ceramic vessel crafted by the player character...",
  "stable_attributes": {
    "description": { "value": "A simple, functional spear used for survival..." },
    "material": { "value": "wood" },
    "purpose": { "value": "hunting, combat, and ceremonial use" },
    "aliases": { "value": ["sturdy bowl", "ceramic vessel", "sturdy spear shafts"] },
    "species": { "value": "ceramic" },
    "race": { "value": "crafted" },
    "class": { "value": "vessel" }
  }
}
```

The entity has `species: "ceramic"`, `race: "crafted"`, contradictory aliases (both bowl and spear), and a description mentioning spear while the identity says bowl. The LLM confused two items and merged them into one entity.

### Abstract concepts classified as physical items

| Entity ID | Name | Created | Issue |
|-----------|------|---------|-------|
| `item-pattern-disruption-method` | "the disruption" | turn-265 | A magical technique/method is not a physical item |
| `item-plague-treatment-protocol` | "Plague Treatment Protocol" | turn-265 | A medical procedure is not a physical item |

These appear in `char-player`'s equipment list:
```json
"equipment": ["Talismans/Fetishes (Necklace)", "Pattern Disruption Method", "Plague Treatment Protocol"]
```

### Vague references persisted as trackable characters

| Entity ID | Name | Issue |
|-----------|------|-------|
| `char-special-someone` | "special someone" | Vague DM reference to an unnamed companion, now tracked as a distinct entity with full status updates through turn-259 |
| `char-figures` | "two figures" | Early-game captors, never given names, but entity persists and accumulates unrelated activities |

### Items with inapplicable schema fields

The LLM fills character-schema fields on items:
- `item-spear`: has `species`, `race`, `class` fields (meaningless for items)
- `item-bundle`: named "Lyrawyn" (a character name applied to an item)

## Root Cause

1. **Identity overwrite**: The detail update step receives entity ID + turn text and rewrites the entity's identity/name. If the LLM misidentifies which item is being referenced, it overwrites the wrong entity. The `_update_existing_entity()` function in `catalog_merger.py` allows name changes without validation.

2. **Abstract-as-item**: The discovery step has no concept filter — if the DM says "your plague treatment protocol", the LLM classifies "Plague Treatment Protocol" as a new item because it appears in a possessive context.

3. **Schema field bleeding**: The detail prompt template includes fields like `species`, `race`, `class` regardless of entity type. The LLM fills them even for items.

## Suggested Fixes

1. **Guard identity/name overwrites**: In `_update_existing_entity()`, reject name changes that differ significantly from the original (e.g. "spear" → "sturdy bowl" should be rejected or flagged for review).

2. **Type-appropriate schema**: Use different detail prompts for items vs characters. Items should not have `species`/`race`/`class` fields.

3. **Abstract concept filter**: Add a classification gate after discovery that rejects "items" whose names match method/protocol/technique patterns, or that have no physical properties.

4. **Name stability check**: If an entity's name has changed >50% from its original (Levenshtein or similar), log a warning and keep the original name.

## Labels

bug

### PR DESCRIPTION
**Pull request by daviburg**
## Summary

Guards against entity identity corruption where LLM hallucinations cause unrelated entities to overwrite each other's names and identities (e.g. "Crude wood-hafted spear" → "sturdy bowl"). Also rejects abstract concepts misclassified as items and strips character-only schema fields from item entities.

## Changes

| File | Change |
|---|---|
| `tools/catalog_merger.py` | Name stability guard: rejects name changes with zero meaningful word overlap |
| `tools/catalog_merger.py` | Identity overwrite guard: rejects identity updates when update name doesn't match current name |
| `tools/semantic_extraction.py` | Abstract-item discovery filter: rejects items named with abstract concept patterns (method, protocol, technique, etc.) |
| `tools/semantic_extraction.py` | `_coerce_entity_fields()`: strips character-only fields (species, race, class, alignment, age) from items |
| `tests/test_identity_guard.py` | 11 new tests covering all four guards |
| `tests/test_catalog_merger_v2.py` | Updated existing test for new guard behavior; added no-overlap rejection test |

## Entities caught by these guards

- **Name corruption**: "Crude wood-hafted spear" → "sturdy bowl" (no word overlap — blocked)
- **Identity corruption**: Identity text from a different entity applied to wrong ID (blocked when names don't overlap)
- **Abstract items**: "Pattern Disruption Method", "Plague Treatment Protocol", "Ancient technique" (filtered at discovery)
- **Schema pollution**: species/race/class/alignment/age fields on item entities (stripped during coercion)

## Test results

1575 passed, 1 skipped, 0 failed (12 new tests added: 11 in `test_identity_guard.py`, 1 in `test_catalog_merger_v2.py`)

Closes #339

### CODE REVIEW

File: tools/semantic_extraction.py [hunk 1]
```diff
@@ -702,6 +702,14 @@ def _coerce_entity_fields(entity_data) -> dict | None:
     if isinstance(vs, dict) and "last_updated_turn" not in vs and has_valid_turn:
         vs["last_updated_turn"] = turn_id
 
+    # Strip character-only schema fields from items (#339)
+    if entity_data.get("type") == "item":
+        _ITEM_INVALID_ATTRS = {"species", "race", "class", "alignment", "age"}
+        sa = entity_data.get("stable_attributes", {})
+        for attr_key in list(sa.keys()):
+            if attr_key.lower() in _ITEM_INVALID_ATTRS:
+                del sa[attr_key]
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/semantic_extraction.py]: **[@developer]** Fixed in a78c6a8: added isinstance(sa, dict) guard before iterating sa.keys() in the item field-stripping block.

File: tools/catalog_merger.py [hunk 2]
```diff
@@ -1117,7 +1117,20 @@ def _update_existing_entity(current: dict, update: dict, *, known_entity_names:
...
+                old_tokens = set(current_name.replace("-", " ").split()) - _trivial
+                new_tokens = set(update_name.replace("-", " ").split()) - _trivial
+                if old_tokens and new_tokens and not (old_tokens & new_tokens):
+                    print(f"  GUARD: rejecting identity update for '{current.get('id')}' — "
+                          f"update name '{update_name}' doesn't match current '{current_name}'",
+                          file=sys.stderr)
+                else:
+                    current["identity"] = update["identity"]
+            else:
+                current["identity"] = update["identity"]
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/catalog_merger.py]: **[@developer]** Fixed in a78c6a8: moved the name-overlap check to the top of _update_existing_entity() and added an early return that skips all content merges (identity, stable_attributes, volatile_state, name, status) when the update name has zero word overlap with the current name. Only last_updated_turn is advanced.

File: tools/semantic_extraction.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/semantic_extraction.py]: **[@tester]** Verified in a78c6a8: `isinstance(sa, dict)` guard is present at line 710 of `semantic_extraction.py`. The `_coerce_entity_fields` function now checks `if isinstance(sa, dict):` before iterating `sa.keys()`. Test `TestCharacterFieldStripping::test_species_removed_from_item` passes. Resolving.

File: tools/catalog_merger.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/catalog_merger.py]: **[@tester]** Verified in a78c6a8: upfront `_name_mismatch` check at line 1117 of `catalog_merger.py` runs before all content merges (identity, stable_attributes, volatile_state) and does an early `return`. Tests confirm: "Crude wood-hafted spear" is protected from "sturdy bowl" rename, "The elder" ??? "Elder Lyra" is allowed (word overlap on "elder"), identity is not overwritten when names mismatch. All 1584 tests pass. Resolving.

**Review by daviburg — COMMENTED** [review]
**[@reviewer]** APPROVE. Both Copilot automated review concerns addressed and verified. Upfront mismatch guard correctly prevents all content merges on zero-overlap name changes. isinstance(sa, dict) guard present. Edge cases (proper name reveals, title additions) correctly handled. 12 new tests with good coverage. Nit: secondary overlap check in name-change section is now dead code (upfront guard catches all zero-overlap cases) — harmless but could be removed in a future cleanup.
